standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -2594,39 +2640,51 @@ public void cleanupRecords(HiveObjectType type, Database db, Table table,
           case DATABASE: {
```

Comment:
can catalog itself be dropped?

Reply:
Yes, but there's no cascading drop if it now, so the drops of DBs in the catalog should handle removing all the locks and transactions.